sklearnex/cluster/k_means.py

```diff
@@ -1,17 +1,368 @@
+                "init": self.init,
+                "max_iter": self.max_iter,
+                "tol": self.tol,
+                "n_init": self.n_init,
+                "verbose": self.verbose,
+                "random_state": self.random_state,
+            }
+
+            self._onedal_estimator = onedal_KMeans(**onedal_params)
+
+        def _onedal_fit_supported(self, method_name, X, y=None, sample_weight=None):
+            assert method_name == "fit"
+
+            class_name = self.__class__.__name__
+            patching_status = PatchingConditionsChain(f"sklearn.cluster.{class_name}.fit")
+
+            sample_count = _num_samples(X)
+            self._algorithm = self.algorithm
+            supported_algs = ["auto", "full", "lloyd", "elkan"]
+            correct_count = self.n_clusters < sample_count
+            is_data_supported = (
+                _is_csr(X) and daal_check_version((2024, "P", 600))
+            ) or not issparse(X)
+            sample_weight = _check_sample_weight(
+                sample_weight, X, dtype=X.dtype if hasattr(X, "dtype") else None
+            )
+
+            patching_status.and_conditions(
+                [
+                    (
+                        self.algorithm in supported_algs,
+                        "Only lloyd algorithm is supported, elkan is computed using lloyd",
+                    ),
+                    (correct_count, "n_clusters is smaller than number of samples"),
+                    (
+                        np.allclose(sample_weight, np.ones_like(sample_weight)),
+                        "Sample weights are not ones.",
+                    ),
+                    (
+                        is_data_supported,
+                        "Supported data formats: Dense, CSR (oneDAL version >= 2024.6.0).",
+                    ),
+                ]
+            )
+
+            return patching_status
+
+        def fit(self, X, y=None, sample_weight=None):
+            if sklearn_check_version("1.0"):
+                self._check_feature_names(X, reset=True)
+            if sklearn_check_version("1.2"):
+                self._validate_params()
+
+            dispatch(
+                self,
+                "fit",
+                {
+                    "onedal": self.__class__._onedal_fit,
+                    "sklearn": sklearn_KMeans.fit,
+                },
+                X,
+                y,
+                sample_weight,
+            )
+
+            return self
+
+        def _onedal_fit(self, X, _, sample_weight, queue=None):
+            X = self._validate_data(
+                X,
+                accept_sparse="csr",
+                dtype=[np.float64, np.float32],
+            )
+
+            if sklearn_check_version("1.2"):
+                self._check_params_vs_input(X)
+            else:
+                self._check_params(X)
+
+            self._n_features_out = self.n_clusters
+            self._n_threads = _openmp_effective_n_threads()
```

Comment:
seems like this _n_threads part can maybe be removed

Reply:
based on sklearn implementation, but not totally sure

Reply:
Seems like it is needed for score function